### ISSUE TITLE
RMSNorm on NJT doesn't dispatch to fused implementation

### ISSUE THREAD

**Issue by schmidt-jake** [pytorch/pytorch#191739]
### 🐛 Describe the bug

`F.rms_norm` on a jagged NestedTensor falls back to a decomposition into elementwise ops rather than reaching `_fused_rms_norm`, even when normalizing purely over trailing dense dims (the ragged dim is untouched). The dense path uses the fused kernel.

### Repro

```python
import time
import torch
import torch.nn.functional as F
from torch.utils._python_dispatch import TorchDispatchMode


class OpLog(TorchDispatchMode):
    def __init__(self):
        self.ops = []

    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
        self.ops.append(func._overloadpacket.__name__)
        return func(*args, **(kwargs or {}))


dim, n = 512, 52424
values = torch.randn(n, dim, device="cuda")
lengths = [149] * (n // 149) + [n - 149 * (n // 149)]
offsets = torch.tensor([0, *torch.tensor(lengths).cumsum(0).tolist()], device="cuda")
njt = torch.nested.nested_tensor_from_jagged(
    values, offsets=offsets, min_seqlen=0, max_seqlen=149
)
weight = torch.ones(dim, device="cuda")

for name, x in (("dense", values), ("njt", njt)):
    with OpLog() as log:
        F.rms_norm(x, (dim,), weight=weight, eps=1e-6)
    print(name, sorted(set(log.ops)))
```

### Related issues and PRs

- #135872 — *Support rms_norm() for NJT*, landed 2024-09-17 (`a8382847f452`). This is the PR that added the `__torch_function__` block quoted above. Its own description says it "SymInt-ifies `rms_norm()`, allowing NJT to use the same decomp", so routing through the decomposition was the original design, not a regression.
- #153666 — *Fused RMSNorm implementation*, added `_fused_rms_norm`; landed 2025-07-22 after two reverts, followed by #159317 (*Fused RMSNorm Housekeeping*, 2025-07-29). It post-dates #135872 by ~10 months, and `_fused_rms_norm` still does not appear anywhere under `torch/nested/`.
- #164289 — *Remove CompositeImplicitAutograd from `_fused_rms_norm`*, closed 2025-10-10 without landing. It reports the same class of problem from the dense/`aot_eager` side: "The failure of bitwise equivalence is because aot_eager decomposes rms_norm." `_fused_rms_norm` still carries a `CompositeImplicitAutograd[alias]` registration in `2.14.0.dev20260707` (checked with `torch._C._dispatch_dump("aten::_fused_rms_norm")`).
- #141916 — *rms_norm forward performance analysis* (open), dense-side `rms_norm` performance.

### Versions

PyTorch version: 2.14.0.dev20260707+cu130
CUDA used to build PyTorch: 13.0
Python version: 3.12.13 (main, Jun 23 2026, 15:18:55) [Clang 22.1.3] (64-bit runtime)
Python platform: Linux-6.11.0-1016-nvidia-x86_64-with-glibc2.39
OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Libc version: glibc-2.39

Is CUDA available: True
GPU models and configuration: GPU 0: NVIDIA H200 (143771 MiB)
Nvidia driver version: 570.195.03
CUDA runtime version: 12.9.86
cuDNN version: 9.10.1

CPU: Intel(R) Xeon(R) Platinum 8468, 128 threads, 2 sockets

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ

**Comment by UlviDemirsoy** [comment]
I can reproduce this on CUDA (`dense` hits `_fused_rms_norm`, `njt` falls through to `mean`/`pow`/`rsqrt`/`mul`).

Proposed fix: in the NJT `rms_norm` `__torch_function__` path, after the existing ragged-dim validation, redispatch to dense `torch.rms_norm` on `input._values` and rewrap with the original jagged metadata. That matches other NJT unary redispatches and lets the existing fused kernel run when normalizing only over trailing dense dims.

Draft PR: https://github.com/pytorch/pytorch/pull/191851

Happy to adjust if you'd prefer a different approach. Could this be marked `actionable` if the direction looks good?

**Comment by schmidt-jake** [comment]
@UlviDemirsoy yep that approach sounds good! and your draft PR lgtm
